doc/generic/pgf/text-en/pgfmanual-en-tutorial.tex

```diff
@@ -1100,11 +1100,16 @@ \subsection{Repeating Things: For-Loops}
 \end{codeexample}
 
 If you provide \emph{two} numbers before the |...|, the |\foreach| statement
-will use their difference for the stepping:
+will use their difference for the stepping, but be wary of floating-point
+rounding errors:
 %
 \begin{codeexample}[]
-\tikz \foreach \x in {-1,-0.5,...,1}
-       \draw (\x cm,-1pt) -- (\x cm,1pt);
+\begin{tikzpicture}
+  \foreach \x in {-1,-0.5,...,1}
+    \draw (\x cm,-1pt) -- (\x cm,1pt);
+  \foreach \x in {-1,-0.8,...,1}
+    \draw (\x cm,-7pt) -- (\x cm,-5pt);
```

Comment:
Step `0.2` doesn't demonstrate the rounding error because it's rounded down:
```tex
 \foreach \x in {-1,-0.8,...,1} {\x, }
% gives
% 0, .2, 0.4, 0.59999, 0.79999, 0.99998,
```

I understand the consideration that `{-1,-0.9,...,1}` would draw too many ticks, so will `{-1,-0.9,...,0}` do the job?
```suggestion
  \foreach \x in {-1,-0.9,...,0}
    % tick at 0 is not drawn due to rounding errors
    \draw (\x cm,-7pt) -- (\x cm,-5pt);
```
![image](https://user-images.githubusercontent.com/6376638/134777718-74247ed2-0697-4744-9e30-a8cc58b68dd9.png)

Reply:
Oops, thanks for the catch. I was initially nervous that a step of 0.1 would be too dense, rather than too many, but I think the added comment helps to draw attention to precisely where the missing tick should be, making the density less problematic. I also think that stopping at 0 rather than 1 makes the number of ticks more easily counted, in case people want to think "how many ticks should be drawn in total"?

Reply:
Ah my poor English failed to properly express "too dense".